x-pack/plugins/infra/server/lib/adapters/nodes/extract_paths_and_metrics.ts

```diff
@@ -18,22 +18,22 @@ import {
 
 interface InfraPathsAndMetricOptions {
   path: InfraPathInput[];
-  metrics: InfraMetricInput[];
+  metric: InfraMetricInput | null;
 }
 
-export function extractPathsAndMetrics(info: GraphQLResolveInfo): InfraPathsAndMetricOptions {
-  if (info.variableValues.metrics && info.variableValues.path) {
+export function extractPathsAndMetric(info: GraphQLResolveInfo): InfraPathsAndMetricOptions {
+  if (info.variableValues.metric && info.variableValues.path) {
     return {
-      metrics: info.variableValues.metrics as InfraMetricInput[],
+      metric: info.variableValues.metric as InfraMetricInput,
       path: info.variableValues.path as InfraPathInput[],
     };
   }
 
-  const { path, metrics }: InfraPathsAndMetricOptions = info.fieldNodes.reduce(parseFieldNodes, {
+  const { path, metric }: InfraPathsAndMetricOptions = info.fieldNodes.reduce(parseFieldNodes, {
```

Comment:
This code path seems to not work correctly. The following query leads to an exception:

```graphql
  query WaffleNodesQuery(
    $sourceId: ID!
    $timerange: InfraTimerangeInput!
    $filterQuery: String
    $path: [InfraPathInput!]!
  ) {
    source(id: $sourceId) {
      id
      map(timerange: $timerange, filterQuery: $filterQuery) {
        nodes(path: $path) {
          path {
            value
          }
          metric(metric: {type:cpu}) {
            name
            value
          }
        }
      }
    }
  }
```

with these variables:

```json
{
  "sourceId": "default",
  "path": [
    {
      "type": "hosts"
    }
  ],
  "timerange": {
    "from": 1539692147700,
    "interval": "5m",
    "to": 1539692747700
  },
  "filterQuery": null
}
```

It works if you don't inline the metric or path arguments because then the previous condition short-circuits this.

Reply:
I'm going to move all the arguments from the sub branches (`nodes` and `metric`) to `map`. This will eliminate the need to parse the info object and fix this endpoint once and for all.

Reply:
That sounds like it would loose us most of the graphql advantages such as querying multiple node groups and metrics via aliasing. I think we should discuss this before throwing out the baby with the bathwater.

Reply:
The advantage is that it simplifies the code base. We don't have any plans to query the GraphQL endpoint with aliases at this point. If we need the ability to do that in the future for some use case then we can put the effort in at that time.